core/orchestrator.go

```diff
@@ -856,35 +873,45 @@ func (rtm *RemoteTranscoderManager) selectTranscoder() *RemoteTranscoder {
 	}
 
 	for checkTranscoders(rtm) {
+		currentTranscoder, sessionExists := rtm.streamSessions[sessionId]
```

Comment:
```suggestion
		currentTranscoder, sessionExists := rtm.streamSessions[sessionId]
		if !sessionExists {
		    last := len(rtm.remoteTranscoders) - 1
		    currentTranscoder = rtm.remoteTranscoders[last]
		}
```

I think this should allow the RemoteTranscoderManager to quickly switch transcoders if a transcoder for an existing stream session is no longer live.

Reply:
Well, it is doing just that in the following lines, after the early return if sessionExists. It would effectively be the same, I suppose.

Reply:
The idea here is always run the live transcoder map check after selecting a transcoder. Previously, if a session already existed, the method was returning the transcoder immediately before running the live transcoder map check.

Reply:
Oh right, this makes sense now, thanks for clarifying, will make the changes.

Reply:
The idea was that we wouldn't have to do this check in `selectTranscoders` but just straight up select a new one. 

But I'm fine with this suggestion.

Reply:
> The idea was that we wouldn't have to do this check in selectTranscoders but just straight up select a new one.

Not sure I follow? I believe what you're describing is what will happen with the suggestion. If a transcoder for an existing session is selected, but is no longer live it will be discarded and another transcoder will be selected.

Reply:
The current implementation would remove the session from `streamSessions` as soon as the connection between O and T closes. 

So we would already select a new transcoder on the first iteration rather than on the second. 

With the suggestion we would first remove the current transcoder that is no longer live in the first iteration, then select a new one in the second iteration. 

But as I mentioned I do see the cleanliness benefit of the suggestion as well so I don't have a strong opinion towards one or the other.

Reply:
Cool. Let's go with the suggestion then.

Reply:
So correct me here: after this PR, we would only be incrementing load on new sessions, so if a segment on the same session comes in, it would still increment the load in the lines below. Also it would error out with `ErrNoTranscodersAvailable` if it as max load, even if it gets another segment from an already existing stream session.
Thoughts? How do you'll recommend I proceed? Do I add an `if (!sessionExists)` around the capacity check and subsequent lines until the return?